external-crates/move/crates/move-compiler/src/hlir/ast.rs

```diff
@@ -574,6 +579,58 @@ impl UnannotatedExp_ {
     pub fn is_unreachable(&self) -> bool {
         matches!(self, UnannotatedExp_::Unreachable)
     }
+
+    /// Returns the set of variables free in the expression.
+    pub fn free_vars(&self) -> BTreeSet<Var> {
+        fn free_vars_recur(set: &mut BTreeSet<Var>, e: &UnannotatedExp_) {
```

Comment:
Do we need our good ol friend the stack_grow (or whatever it is called) here?

Reply:
I think not, since expressions are pretty shallow after HLIR translation, but I rewrote it as a work queue just in case.